islandora.module

```diff
@@ -331,6 +333,175 @@ function islandora_form_alter(&$form, FormStateInterface $form_state, $form_id)
       }
     }
   }
+  $form_object = $form_state->getFormObject();
+  $utils = \Drupal::service('islandora.utils');
+
+  if ($form_object instanceof EntityForm) {
+    $entity = $form_object->getEntity();
+
+    if ($entity->getEntityTypeId() == "node" && $utils->isIslandoraType($entity->getEntityTypeId(), $entity->bundle()) && strpos($form['#form_id'], 'delete_form') !== FALSE) {
```

Comment:
Given this is specifically targeting node, and furthermore, the `EntityForm` of it, should this instead be more targetedly adjusting the definition of `node` content entities, such as substituting our own [`delete`](https://git.drupalcode.org/project/drupal/-/blob/f18b610de50a2d1a01e338f8be71b2478a0be51b/core/modules/node/src/Entity/Node.php#L35) and [`delete-multiple-confirm`](https://git.drupalcode.org/project/drupal/-/blob/f18b610de50a2d1a01e338f8be71b2478a0be51b/core/modules/node/src/Entity/Node.php#L37) forms?

In any case, I almost feel like this behaviour probably should be behind a feature toggle, or in another (sub)module, as something like the [`entity_reference_integrity`](https://www.drupal.org/project/entity_reference_integrity) module with its `entity_reference_integrity_enforce` submodule should prevent this functionality from working.

Reply:
in [5c24c19](https://github.com/Islandora/islandora/pull/905/commits/5c24c190184b680914f10bae7fbff23fa33663bb) a toggle was added.